onnxruntime/core/providers/webgpu/math/einsum.cc

```diff
@@ -34,13 +34,13 @@ bool IsInteger(const std::string& s) {
 }
 }  // namespace
 
-#define WEBGPU_EINSUM_TYPED_KERNEL_DECL(version)                                               \
-  ONNX_OPERATOR_TYPED_KERNEL_EX(                                                               \
-      Einsum, kOnnxDomain, version, float, kWebGpuExecutionProvider,                           \
-      (*KernelDefBuilder::Create()).TypeConstraint("T", DataTypeImpl::GetTensorType<float>()), \
+#define WEBGPU_EINSUM_KERNEL_DECL(version)                                               \
+  ONNX_OPERATOR_KERNEL_EX(                                                               \
+      Einsum, kOnnxDomain, version, kWebGpuExecutionProvider,                            \
+      (*KernelDefBuilder::Create()).TypeConstraint("T", WebGpuSupportedFloatTypes()),    \
```

Comment:
Can you relax it to WebGpuSupportedNumberTypes if all related tests can pass?

Reply:
Jiajia, there is not uint32 case and only one int32 case, if relaxing the types, I am not sure whether there will be some uncovered errors.